src/fsharp/IlxGen.fs

```diff
@@ -6294,15 +6294,17 @@ and GenTypeDef cenv mgbuf lazyInitInfo eenv m (tycon:Tycon) =
              [ for fspec in tycon.AllFieldsAsList do
 
                    let useGenuineField = useGenuineField tycon fspec
-                   
+
                    // The property (or genuine IL field) is hidden in these circumstances:  
                    //     - secret fields apart from "__value" fields for enums 
                    //     - the representation of the type is hidden 
                    //     - the F# field is hidden by a signature or private declaration 
-                   let isPropHidden = 
-                       ((fspec.IsCompilerGenerated && not tycon.IsEnumTycon) ||
-                        hiddenRepr ||
-                        IsHiddenRecdField eenv.sigToImplRemapInfo (tcref.MakeNestedRecdFieldRef fspec))
+                   let isPropHidden =
+                        // Enums always have public cases irrespective of Enum Visibility
+                        if tycon.IsEnumTycon then false
```

Comment:
The `value__` field in enums should always be hidden, I think this will have changed?.  

Also we need to have a CodeGen test for enums...

Reply:
Yes I expect it will.  I can fix those.

Reply:
@dsyme 

The C# compiler emits a public value__ field

![image](https://user-images.githubusercontent.com/5175830/40563340-41c6a080-6019-11e8-8788-9c40cd90e1c9.png)


I think we should match C# right?